submissions/lalalilulu/tiny-world-pre-oop/index.js

```diff
@@ -0,0 +1,98 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+
+    constructor(species, name, gender, saying, friends= [], legs) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+        this.legs = legs;
+        this.friends = friends.join(", ");
+    }
+
+    toString() {
+        return`${this.saying} I am a ${this.species}, my name is ${this.name}, my gender is ${this.gender}. I have ${this.legs} legs. My friends: ${this.friends}`;
+    }
+}
+
+class Animal extends Inhabitant{
+
+    constructor(species, name, gender, saying, friends, legs = 4) {
+        super(species, name, gender, saying, friends, legs);
+    }
+}
+
+class Human extends Animal{
+
+    constructor(name, gender, saying, friends, legs = 2, hands = 2) {
+        super('human', name, gender, saying, friends, legs);
+        this.hands = hands;
+    }
+
+    toString() {
+        return super.toString() + ` and I also have ${this.hands} hands`;
+    }
+}
+
+class Dog extends Animal{
+
+    constructor(name, gender, saying, friends) {
+        super('dog', name, gender, saying, friends, 4);
+
+    }
+}
+
+class Cat extends Animal{
+
+    constructor(name, gender, saying, friends) {
+        super('cat', name, gender, saying, friends, 4);
+
+    }
+}
+
+class CatWoman extends Animal{
+
+    constructor(name, saying, friends) {
+        super('cat', name, 'female', saying, friends, 4);
```

Comment:
It is all like a `Cat`: cat species, 4 legs, no hands.
Doesn't look like good reasons to have a dedicated class for this.
Good reasons would be if CatWoman was all like a human with the exception that it speaks like a cat.